src/handler.rs

```diff
@@ -0,0 +1,183 @@
+//! STAR Randomness web service route implementation
+
+use axum::extract::{Json, State};
+use axum::http::StatusCode;
+use base64::prelude::{Engine as _, BASE64_STANDARD as BASE64};
+use serde::{Deserialize, Serialize};
+use tracing::debug;
+
+use ppoprf::ppoprf;
+use std::sync::{Arc, RwLock};
+
+use crate::Config;
+
+/// Internal state of the OPRF service
+pub struct OPRFServer {
+    /// oprf implementation
+    pub server: ppoprf::Server,
+    /// currently-valid randomness epoch
+    pub epoch: u8,
+    /// RFC 3339 timestamp of the next epoch rotation
+    pub next_epoch_time: Option<String>,
+}
+
+/// Shareable wrapper around the server state
+pub type OPRFState = Arc<RwLock<OPRFServer>>;
+
+impl OPRFServer {
+    /// Initialize a new OPRFServer state with the given configuration
+    pub fn new(config: &Config) -> Result<Self, ppoprf::PPRFError> {
+        // ppoprf wants a vector, so generate one from our range.
+        let epochs: Vec<u8> =
+            (config.first_epoch..=config.last_epoch).collect();
+        let epoch = epochs[0];
```

Comment:
That's a nice idea. I filed issue #54 to follow up.

Reply:
i would argue that this is essential for prod rather than just a "nice to have". the client depends on the epochs being in a sequential order, in order to delete expired measurements